app/views/admin/events/_all_with_comments.csv.haml

```diff
@@ -0,0 +1,23 @@
+- headers = ['Event ID',
+             'Title',
+             'Abstract',
+             'Start time',
+             'Submitter',
+             'Speaker',
+             'Event Type',
+             'Track',
+             'Difficulty Level',
+             'Room',
+             'State',
+             'Comments']
+= CSV.generate_line ["Events with Comments"]
+= CSV.generate_line headers
+- @events.each do |event|
+  - all_comments = ''
+  - event.root_comments.each do |comment|
+    - all_comments << "#{comment.created_at.strftime("%Y-%m-%d")} #{comment.created_at.strftime("%I:%M%p")}  #{comment.user.name}: #{comment.body}\n"
+  = CSV.generate_line([event.id, event.title, event.abstract, (event.time.present? ? "#{event.time.strftime("%Y-%m-%d")} #{event.time.strftime("%I:%M%p")} " : 'nil'),
+                       event.submitter.name, event.speaker_names, event.event_type.title,
+                       (event.track.present? ? event.track.name : 'nil'),
+                       (event.difficulty_level.present? ? event.difficulty_level.title : 'nil'),
+                       (event.room.present? ? event.room.name : 'nil'), event.state, all_comments]).html_safe
```

Comment:
@differentreality  In all event with comments part of csv, first I am extracting all comments in an array then using in csv file generation. If that array is empty,  it is messing up other columns.  I added html_safe to avoid that mess.  For your reference:
[with out html_safe](http://i.imgur.com/0zsSffM.png)
[with html safe](http://i.imgur.com/LXNKjnO.png)
We are not adding an array in  all events and confirmed events - that's why you don't see html_safe.